src/PreOrder.sol

```diff
@@ -98,11 +98,11 @@ contract PreOrder is
     // Mints a token with ETH as payment
     function mint(uint8 _tier) payable external whenNotPaused {
         require(msg.value == tiers[_tier].costWei, "Incorrect amount sent");
-        require(tiers[_tier].mintCount < tiers[_tier].maxSupply, "Tier sold out");
 
         (bool success, ) = gnosisSafe.call{value: msg.value}("");
         require(success, "Transfer failed");
 
+        require(tiers[_tier].mintCount < tiers[_tier].maxSupply, "Tier sold out");
```

Comment:
Can you consider applying this pattern? 
this is useful to handle these re-entrancy issues

https://docs.soliditylang.org/en/v0.6.11/security-considerations.html#use-the-checks-effects-interactions-pattern

you can just move the gnosisSafe.call to the end of the function